packages/kit/src/runtime/client/router.js

```diff
@@ -169,20 +196,48 @@ export class Router {
 				return;
 			}
 
-			const noscroll = a.hasAttribute('sveltekit:noscroll');
-			this._navigate(url, noscroll ? scroll_state() : null, false, [], url.hash, {}, 'pushState');
-			event.preventDefault();
+			this._navigate({
+				url,
+				scroll: a.hasAttribute('sveltekit:noscroll') ? scroll_state() : null,
+				keepfocus: false,
+				chain: [],
+				details: {
+					state: {},
+					replaceState: false
+				},
+				accepted: () => event.preventDefault(),
+				blocked: () => event.preventDefault()
+			});
 		});
 
 		addEventListener('popstate', (event) => {
```

Comment:
It is very helpful to know if we're navigating because of a `popstate` event, so we can close open menus and modals on browser back button use, before navigating away from a route.

Can we thread this "back button used" bit of info through `_navigate` all the way to `beforeNavigate`?